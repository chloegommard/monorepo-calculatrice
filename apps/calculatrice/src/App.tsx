import { ToastContainer } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'
import { useState, useEffect } from 'react'
import {
  GlobalStyle,
  StyledGridContainer,
  StyledScreen,
  StyledButton,
  StyledLink,
} from './styles'
import { solve } from './helpers'

const buttonValues: string[] = [
  '1',
  '2',
  '3',
  '+',
  '4',
  '5',
  '6',
  'â€”',
  '7',
  '8',
  '9',
  '*',
  '0',
  '.',
  '-',
  '/',
  '%',
  '^',
]

export type NumberOrEmptyString = number | ''

function App(): JSX.Element {
  // states
  const [result, setResult] = useState<NumberOrEmptyString>('')
  const [screenValue, setScreenValue] = useState<string>('')

  // functions
  const display = (number: string): void => {
    setScreenValue(`${screenValue}${number}`)
  }

  const clear: VoidFunction = () => {
    setScreenValue('')
  }

  useEffect(() => {
    setScreenValue(String(result))
  }, [result])

  return (
    <>
      <GlobalStyle />
      <ToastContainer />
      <StyledGridContainer data-testid="calculator">
        <StyledScreen value={screenValue} readOnly id="screen" />
        {buttonValues.map(button => (
          <StyledButton
            key={button}
            id={button}
            onClick={() => display(button)}
          >
            {button}
          </StyledButton>
        ))}
        <StyledButton
          id="equal"
          onClick={() => {
            clear()
            solve(screenValue, setResult)
          }}
        >
          =
        </StyledButton>
        <StyledButton id="clear" onClick={clear}>
          C
        </StyledButton>
      </StyledGridContainer>
      <StyledLink href="http://localhost:3000/docs/get-started/intro">
        Documentation
      </StyledLink>
    </>
  )
}

export default App
