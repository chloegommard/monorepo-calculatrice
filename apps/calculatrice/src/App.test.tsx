import { expect, describe, it } from 'vitest'
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import App from './App'

describe('Visual regression', () => {
    it('calculator', () => {
      // Arrange
      // Mount / render the App
      render(<App />)
      
      // Act
      // Do the actions, if any...
  
      // Assert
      // Test if the snapshot is the same
      // const button = getByText('1');

      const calc = screen.getByTestId('calculator')
      expect(calc).toMatchSnapshot()
    })
    
  })