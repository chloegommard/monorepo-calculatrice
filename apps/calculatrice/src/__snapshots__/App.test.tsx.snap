// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Visual regression > calculator 1`] = `
.c0 {
  display: grid;
  justify-items: stretch;
  grid-template-columns: repeat(4, 60px [col-start]);
  grid-template-rows: repeat(6, 60px [row-start]);
  box-shadow: 5px 5px 5px rgb(217, 212, 212);
}

.c2 {
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: Cascadia Code;
  background-color: #f9f9f9;
  cursor: pointer;
  transition: border-color 0.25s;
  justify-self: stretch;
}

.c2:hover {
  border-color: #646cff;
}

.c2:focus,
.c2:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

.c1 {
  font-family: Cascadia Code;
  grid-row-start: 1;
  grid-column-start: 1;
  grid-column-end: 5;
  font-size: 1.5em;
  font-weight: 500;
  color: #abddad;
  background-color: #1a1a1a;
  cursor: pointer;
  justify-self: stretch;
}

<div
  class="c0"
  data-testid="calculator"
>
  <input
    class="c1"
    id="screen"
    readonly=""
    value=""
  />
  <button
    class="c2"
    id="1"
  >
    1
  </button>
  <button
    class="c2"
    id="2"
  >
    2
  </button>
  <button
    class="c2"
    id="3"
  >
    3
  </button>
  <button
    class="c2"
    id="+"
  >
    +
  </button>
  <button
    class="c2"
    id="4"
  >
    4
  </button>
  <button
    class="c2"
    id="5"
  >
    5
  </button>
  <button
    class="c2"
    id="6"
  >
    6
  </button>
  <button
    class="c2"
    id="—"
  >
    —
  </button>
  <button
    class="c2"
    id="7"
  >
    7
  </button>
  <button
    class="c2"
    id="8"
  >
    8
  </button>
  <button
    class="c2"
    id="9"
  >
    9
  </button>
  <button
    class="c2"
    id="*"
  >
    *
  </button>
  <button
    class="c2"
    id="0"
  >
    0
  </button>
  <button
    class="c2"
    id="."
  >
    .
  </button>
  <button
    class="c2"
    id="-"
  >
    -
  </button>
  <button
    class="c2"
    id="/"
  >
    /
  </button>
  <button
    class="c2"
    id="%"
  >
    %
  </button>
  <button
    class="c2"
    id="^"
  >
    ^
  </button>
  <button
    class="c2"
    id="equal"
  >
    =
  </button>
  <button
    class="c2"
    id="clear"
  >
    C
  </button>
</div>
`;
